pytest-selenium/tests/test_search.py

```diff
@@ -129,3 +136,168 @@ def test_TOC_closed_if_search_sidebar_is_displayed(selenium, base_url, book_slug
 
     # AND: Search sidebar is displayed
     assert search_sidebar.is_displayed
+
+
+@markers.test_case("C543239")
+@markers.parametrize("page_slug", ["preface"])
+@markers.nondestructive
+def test_opening_TOC_closes_search_sidebar(selenium, base_url, book_slug, page_slug):
+    """Opening TOC closes search sidebar and content stays in the same location"""
+
+    # GIVEN: Book page is loaded
+    book = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+
+    # Skip any notification/nudge popups
+    while book.notification_present:
+        book.notification.got_it()
+
+    while book.highlighting_CTA_present:
+        book.highlighting_CTA.got_it()
+
+    toolbar = book.toolbar
+    mobile = book.mobile_search_toolbar
+    toc_sidebar = book.sidebar
+    search_sidebar = book.search_sidebar
+    search_term = utility.get_search_term(book_slug)
+
+    if book.is_desktop:
+        # WHEN: Search sidebar is displayed with search results
+        toolbar.search_for(search_term)
+        assert search_sidebar.search_results_present
+
+        # Loop through the words in search term and assert if atleast one of them is highlighted in the book
+        split_search_term = re.findall(r"\w+", search_term)
+        for x in split_search_term:
+            focussed_search_term = book.content.find_elements(By.XPATH, XPATH_SEARCH.format(term=x))
```

Comment:
It's considered a good practice in Python to use variable names that have meaning. So 'x' is a good name when dealing with a coordinate grid, but not as useful in general.